keras_rs/src/layers/embedding/distributed_embedding_test.py

```diff
@@ -318,6 +319,37 @@ def test_model_fit(self, input_type, use_weights):
         with self._strategy.scope():
             layer = distributed_embedding.DistributedEmbedding(feature_configs)
 
+        # Mark ragged inputs as sparse in JAX for the purposes
```

Comment:
Hmm... I don't think it's a good idea, what if you don't mark it sparse either? Does it work?

Reply:
I had reservations about this too, and don't really want users to do the same thing.

If you don't mark it as sparse you get warnings about the actual model inputs not matching the original functional model definition.

For ragged inputs and no explicit weights, when we densify we need to create a new weights tensor to mask out the padding.  We end up in a situation where the functional model expects inputs without weights, but after `preprocess`, we do have a weights tensor.

If you think it's okay to remove the check from `keras.KerasTensor`, then we only use the `ragged` attribute here, that's probably best from a user perspective.

Reply:
> If you don't mark it as sparse you get warnings about the actual model inputs not matching the original functional model definition.

What's the warning?

> We end up in a situation where the functional model expects inputs without weights, but after preprocess, we do have a weights tensor.

I'm not following how that is related to ragged vs. not.

> If you think it's okay to remove the check from keras.KerasTensor, then we only use the ragged attribute here, that's probably best from a user perspective.

Which check? I think we shouldn't change the check in `KerasTensor` but change jax's `SUPPORTS_RAGGED_TENSORS` to True.